### PR TITLE
Abstraction

### PR DESCRIPTION
**Pull request by korkoza**
Hello my friend!

Please take a look to the changes

### CODE REVIEW

File: src/lesson1/AbstractAuto.java [hunk 1]
```diff
@@ -17,40 +17,41 @@ public String getMake() {
...
     }
 
     public void setEngine(String engine) {
         this.engine = engine;
     }
 
-    public void drive(){
-        System.out.println("It can drive using fossil fuel");
+    @Override
+    public void drive() {
```
Comment (toleksyn) [inline src/lesson1/AbstractAuto.java:49]: you do not need this

File: src/lesson1/AnotherClass.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package lesson1;
+
+public class AnotherClass {
+    public static void main(String[] args){
+        Hybrid car1 = new Hybrid("Toyota", "Prius", 2018, "Electricity, Gas");
```
Comment (toleksyn) [inline src/lesson1/AnotherClass.java:5]: variables names should be names specifically

File: src/lesson2/Employee.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    public String getID() {
+        return ID;
+    }
+
+    public void setID(String ID) {
+        this.ID = ID;
+    }
+
+    public abstract float averageSalary();
```
Comment (toleksyn) [inline src/lesson2/Employee.java:38]: getCalculatedSalary()

File: src/lesson2/EmployeeFixed.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package lesson2;
+
+public class EmployeeFixed extends Employee{
```
Comment (toleksyn) [inline src/lesson2/EmployeeFixed.java:3]: FixedSalaryEmployee

File: src/lesson2/EmployeeFixed.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    Float fixedValue;
```
Comment (toleksyn) [inline src/lesson2/EmployeeFixed.java:4]: access modifiers

File: src/lesson2/EmployeeRate.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package lesson2;
+
+public class EmployeeRate extends Employee{
+    float rate;
```
Comment (toleksyn) [inline src/lesson2/EmployeeRate.java:4]: same

File: src/lesson2/EmployeeRate.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+    public EmployeeRate(String name, String surname, String ID, float rate) {
+        super(name, surname, ID);
+        this.rate = rate;
+    }
+
+    @Override
+    public float averageSalary() {
+        return (float) (20.8 * 8 * rate);
```
Comment (toleksyn) [inline src/lesson2/EmployeeRate.java:13]: 20.8, 8 -> parameters

File: src/lesson2/ShowInfo.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package lesson2;
+
+public class ShowInfo {
+    public static void main(String [] args){
+        EmployeeFixed emp1 = new EmployeeFixed("Jhon", "Doe", "1", 3123.8f);
```
Comment (toleksyn) [inline src/lesson2/ShowInfo.java:5]: rename variable names

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/lesson2/EmployeeRate.java (hunk 7 above)
Comment (toleksyn) [reply src/lesson2/EmployeeRate.java:13]: getCalculatedSalary(20.8, 8)

File: src/lesson1/AbstractAuto.java (hunk 1 above)
Comment (korkoza) [reply src/lesson1/AbstractAuto.java]: removed the method from abstract class

File: src/lesson1/AnotherClass.java (hunk 2 above)
Comment (korkoza) [reply src/lesson1/AnotherClass.java]: variables were renamed

File: src/lesson2/Employee.java (hunk 3 above)
Comment (korkoza) [reply src/lesson2/Employee.java]: renamed the method

File: src/lesson2/EmployeeFixed.java (hunk 4 above)
Comment (korkoza) [reply src/lesson2/EmployeeFixed.java]: renamed the class name

File: src/lesson2/EmployeeFixed.java (hunk 5 above)
Comment (korkoza) [reply src/lesson2/EmployeeFixed.java]: added access modifiers

File: src/lesson2/EmployeeRate.java (hunk 6 above)
Comment (korkoza) [reply src/lesson2/EmployeeRate.java]: added access modifiers

File: src/lesson2/EmployeeRate.java (hunk 7 above)
Comment (korkoza) [reply src/lesson2/EmployeeRate.java]: values are passed as parameters now

File: src/lesson2/ShowInfo.java (hunk 8 above)
Comment (korkoza) [reply src/lesson2/ShowInfo.java]: variables were renamed

**Comment by korkoza** [comment]
Hello again
Please have a look to the changes

**Pull request merged**